### ISSUE TITLE
Mapping of QCheckbox in Python3 and PyQt5 is always False

### ISSUE THREAD

**Issue by stlehmann** [mfitzp/pyqtconfig#6]
Hello,

first I want to say that I really like the idea of this project to allow handy mapping of Widgets and Settings. 

Nonetheless I have an issue with the usage of `QCheckBox` with `QSettingsManager`. I' m using Python3.4 with PyQt5. If I uncheck my `QCheckBox` and restart my application it will start checked all the time.

I browsed a bit through the code and found that I have to supply a **bool** default value so that the `_get()` function of `QSettingsManager` can cast the value returned by `QSettings().value()` to bool. 

I fixed this in my code:

``` python
self.settings = QSettingsManager()
self.settings.set_default('plot/autoscale', True)
self.autoscaleCheckbox = QCheckBox(self.tr("autoscale"))
self.settings.add_handler('plot/autoscale', self.autoscaleCheckbox)

```

Still it didn' t work. So I had a look in the `_get()` function of `QSettingsMapper` and found this condition on line 941:

``` python
    elif vt != dt and vt == unicode:
```

This condition is never True in Python3 as unicode has been declare as a function in line 144.

``` python
try:
    # Python2.7
    unicode
except:
    # Python3 recoding
    def unicode(s):
        if isinstance(s, bytes):
            return s.decode('utf-8')
        else:
            return s
```

So for Python3 all Strings are unicode so the condition should look like this:

``` python
    elif vt != dt and vt == str:
```

I' m aware that this solution wouldn' t work in Python2 but I have no other idea at the moment.

Greetings
Stefan

**Comment by stlehmann** [comment]
OK, here two ideas to solve the issue:
1. Use six library 

``` python
from six import string_types
isinstance(s, string_types)
```
1. Check for basestring instead of unicode with try except

``` python
try:
  basestring
except NameError:
  basestring = str
...
elif vt != dt and vt == basestring:
```

**Comment by mfitzp** [comment]
Thanks for the bug report @MrLeeh 

The `basestring` fix should do it, but the `unicode` is used elsewhere for mapping over types. I think the solution would be to use `basestring` as you suggest for the comparison, but also create a function for mapping called `convert_to_str` to be used elsewhere.

**Comment by stlehmann** [comment]
The `basestring` solution works for me. I tried it in a separate branch. Although I made a pull request for replacing some of the many `type()` comparisons by `isinstance()` which allows a more comprehensive type comparison (e.g. inheritance).

**Comment by mfitzp** [comment]
Merged PR #10
